src/NuGet.Clients/NuGet.PackageManagement.UI/Options/ConfigPathsWindowViewModel.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Services.Common;
+using NuGet.Common;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.UI.Options
+{
+    public class ConfigPathsWindowViewModel
+    {
+        public ObservableCollection<ConfigPathsViewModel> ConfigPaths { get; private set; }
+
+        public void SetConfigPaths()
+        {
+            IComponentModel componentModelMapping = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            var settings = componentModelMapping.GetService<Configuration.ISettings>();
+            IReadOnlyList<string> configPaths = settings.GetConfigFilePaths().ToList();
+            ConfigPaths.AddRange(CreateViewModels(configPaths));
+        }
+
+        private ObservableCollection<ConfigPathsViewModel> CreateViewModels(IReadOnlyList<string> configPaths)
+        {
+            var configPathsCollection = new ObservableCollection<ConfigPathsViewModel>();
+            foreach (var configPath in configPaths)
+            {
+                var viewModel = new ConfigPathsViewModel(configPath);
+                configPathsCollection.Add(viewModel);
+            }
+
+            return configPathsCollection;
+        }
+
+        public void OpenConfigFile(ConfigPathsViewModel selectedPath)
+        {
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            var projectContext = componentModel.GetService<INuGetProjectContext>();
+            if (!File.Exists(selectedPath.ConfigPath))
```

Comment:
I assume this line of code `File.Exists` is executed on UI thread. FYI. This operation is not cheap and caused UI delays in the past for other components inside Visual Studio.

Reply:
If you follow the codepath that's calling the file open you'll see that we already do that, https://github.com/NuGet/NuGet.Client/blob/4187eab2d176d6fccb6f0cfc45fc71fd2e750708/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VsCommonOperations.cs#L36-L51. 

So this File.Exists would happen anyways, so it's probably not adding anything perf wise. 

Another note for performance related things is that we shouldn't be too worried about paths that aren't hot. You can't always dodge doing work on the UI thread. 

This will only be triggered when someone *clicks*. Their focus is already there and the 1ms that the check could take is negligible.

Reply:
I'm checking if the file exists here to handle an edge case where a user has the options window open, selects a file, deletes or moves the selected file, then tries to open that same file. In that case, `fullPath` is not null (and can't be null since this command can only be executed when there is a `selectedPath`), so `OpenFile()` doesn't throw an exception. At the same time, the `if (dte.ItemOperations != nulll && File.Exists(fullPath))` condition is not met, since the file no longer exists at that location, so nothing happens. As a result, the user doesn't get any sort of error message when this occurs.

I'll add a comment explaining this.